index.js

```diff
@@ -6,25 +6,23 @@ const kObj = Symbol('object')
 
 module.exports = fp(function (fastify, options, next) {
   var key
+  var salt
   if (options.secret) {
     if (Buffer.byteLength(options.secret) < 32) {
       return next(new Error('secret must be at least 32 bytes'))
     }
 
     key = Buffer.allocUnsafe(sodium.crypto_secretbox_KEYBYTES)
-
-    // static salt to be used for key derivation, not great for security,
-    // but better than nothing
-    var salt = 'mq9hDxBVDbspDR6nLfFT1g=='
+    salt = Buffer.alloc(sodium.crypto_pwhash_SALTBYTES)
 
     if (options.salt) {
-      salt = options.salt
+      salt = Buffer.from(options.salt, 'ascii')
+    } else {
+      sodium.randombytes_buf(salt)
```

Comment:
Unfortunately this is not correct. Multiple instances of the same app must share the same salt.